atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/CommitTsLoaderTest.java

```diff
@@ -88,4 +95,26 @@ public void loadShouldThrowIfANullIsToBeReturned() throws Exception {
 
         assertThat(loader.load(VALID_START_TIMESTAMP)).isEqualTo(ROLLBACK_TIMESTAMP);
     }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void warmingCacheShouldNotPlaceUndueLoadOnTransactionService() throws Exception {
+        long valuesToInsert = 1_000_000;
+
+        doAnswer((invocation) -> {
+            Collection<Long> timestamps = ((Collection<Long>) invocation.getArguments()[0]);
+            if (timestamps.size() > 5_000) {
+                fail("Requested more timestamps in a batch than is reasonable!");
+            }
+            return timestamps.stream().collect(Collectors.toMap(n -> n, n -> n));
+        }).when(mockTransactionService).get(any());
+
+        TLongSet initialTimestamps = new TLongHashSet();
+        initialTimestamps.addAll(LongStream.range(0, valuesToInsert).boxed().collect(Collectors.toList()));
+
+        CommitTsLoader loader = CommitTsLoader.create(mockTransactionService, initialTimestamps);
+        assertThat(loader.load(valuesToInsert - 1)).isEqualTo(valuesToInsert - 1);
+
+        verify(mockTransactionService, atLeast(2)).get(any());
```

Comment:
Is this trying to avoid assuming the 5000 batch size? It's already checked explicitly above. I'd prefer to just make the batch size protected and refer to it directly.

Reply:
:+1: will change this